source/portable/windows/core_pkcs11_pal.c

```diff
@@ -240,8 +294,8 @@ CK_OBJECT_HANDLE PKCS11_PAL_FindObject( uint8_t * pLabel,
 /*-----------------------------------------------------------*/
 
 CK_RV PKCS11_PAL_GetObjectValue( CK_OBJECT_HANDLE xHandle,
-                                 uint8_t ** ppucData,
-                                 uint32_t * pulDataSize,
+                                 CK_BYTE_PTR * ppucData,
+                                 CK_ULONG_PTR pulDataSize,
                                  CK_BBOOL * pIsPrivate )
 {
     CK_RV ulReturn = CKR_OK;
```

Comment:
maintainability: return values should be initialized to failure:

```
CK_RV ulReturn = CKR_KEY_HANDLE_INVALID;
```

This will also let you delete lines 373-376 by slightly altering 359-365:

```
       if( 0 == ulReturn )
        {
            ulReturn = CKR_OK;
            if( ulSize != *pulDataSize )
            {
                ulReturn = CKR_FUNCTION_FAILED;
            }
        }

```

Reply:
Done 👍